client/finality-grandpa/src/lib.rs

```diff
@@ -203,6 +206,9 @@ type CommunicationOutH<Block, H> = finality_grandpa::voter::CommunicationOut<
 	AuthorityId,
 >;
 
+// WIP: convert to struct
+pub type SharedVoterState<Id> = Arc<RwLock<Option<Box<dyn voter::VoterState<Id> + Sync + Send>>>>;
```

Comment:
This shouldn't be typed by `Id` and instead should fix the `VoterState` to `AuthorityId`.